Source/SwiftLintBuiltInRules/Rules/Style/TrailingClosureRule.swift

```diff
@@ -18,127 +19,78 @@ struct TrailingClosureRule: OptInRule {
-                return true
-            }
-
-            return arguments.isEmpty
-        }()
-        // check if there's only one unnamed parameter that is a closure
-        if argumentsCountIsExpected,
-            let offset = dictionary.offset,
-            let totalLength = dictionary.length,
-            let nameOffset = dictionary.nameOffset,
-            let nameLength = dictionary.nameLength,
-            case let start = nameOffset + nameLength,
-            case let length = totalLength + offset - start,
-            case let byteRange = ByteRange(location: start, length: length),
-            let range = file.stringView.byteRangeToNSRange(byteRange),
-            let match = regex("\\s*\\(\\s*\\{").firstMatch(in: file.contents, options: [], range: range)?.range,
-            match.location == range.location {
-            return shouldTrigger()
+        override func visit(_ node: ForStmtSyntax) -> SyntaxVisitorContinueKind {
+            .skipChildren
         }
-
-        return false
     }
+}
 
-    private func filterClosureArguments(_ arguments: [SourceKittenDictionary],
-                                        file: SwiftLintFile) -> [SourceKittenDictionary] {
-        return arguments.filter { argument in
-            guard let bodyByteRange = argument.bodyByteRange,
-                let range = file.stringView.byteRangeToNSRange(bodyByteRange),
-                let match = regex("\\s*\\{").firstMatch(in: file.contents, options: [], range: range)?.range,
-                match.location == range.location
-            else {
-                return false
-            }
-
-            return true
-        }
+private extension SyntaxProtocol {
+    var isNotNextToLeftBrace: Bool {
+        nextToken(viewMode: .sourceAccurate)?.tokenKind != .leftBrace
     }
+}
 
-    private func isAlreadyTrailingClosure(dictionary: SourceKittenDictionary, file: SwiftLintFile) -> Bool {
-        guard let byteRange = dictionary.byteRange,
-            let text = file.stringView.substringWithByteRange(byteRange)
-        else {
-            return false
-        }
+private extension FunctionCallExprSyntax {
+    var lastArgumentIsClosure: Bool {
+        arguments.last?.expression.is(ClosureExprSyntax.self) == true
+    }
 
-        return !text.hasSuffix(")")
+    var containsOnlySingleMutedParameter: Bool {
+        arguments.onlyElement?.isMutedClosure == true
     }
 
-    private func isAnonymousClosureCall(dictionary: SourceKittenDictionary, file: SwiftLintFile) -> Bool {
-        guard let byteRange = dictionary.byteRange,
-            let range = file.stringView.byteRangeToNSRange(byteRange)
-        else {
-            return false
-        }
+    var shouldTrigger: Bool {
+        // If at least last two arguments were ClosureExprSyntax, a violation should not be triggered.
+        arguments.count <= 1
+        || !arguments.dropFirst(arguments.count - 2).allSatisfy({ $0.expression.is(ClosureExprSyntax.self) })
```

Comment:
If I understand it correctly, this method only makes sense together with `node.lastArgumentIsClosure` being checked in front. So you should better combine both into one.

Reply:
Fixed at 6b1d0f9